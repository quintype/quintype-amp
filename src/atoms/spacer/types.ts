export interface SpacerProps {
  token: string;
  align?: SpaceDirection;
}

export type SpaceDirection = "vertical" | "horizontal";
