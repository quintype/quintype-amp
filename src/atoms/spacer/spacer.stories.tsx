import * as React from "react";
import { storiesOf } from "@storybook/react";
import { Spacer } from "./spacer";
import { config, textStory } from "../../__fixtures__";
import { Layout } from "../layout";

storiesOf("Spacer", module)
  .addDecorator((story) => (
    <Layout story={textStory} config={config}>
      {story()}
    </Layout>
  ))
  .add("Vertical spacer", () => (
    <div>
      <div style={{ background: "rebeccapurple", color: "whitesmoke" }}>Some Content</div>
      <Spacer token="xs" />
      <div style={{ background: "rebeccapurple", color: "whitesmoke" }}>Some Content</div>
    </div>
  ))
  .add("Horizontal Spacer", () => (
    <div style={{ display: "flex" }}>
      <div style={{ background: "rebeccapurple", color: "whitesmoke" }}>Some Content</div>
      <Spacer token="xl" align="horizontal" />
      <div style={{ background: "rebeccapurple", color: "whitesmoke" }}>Some Content</div>
    </div>
  ));
