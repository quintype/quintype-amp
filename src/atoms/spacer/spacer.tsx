import React from "react";
import { SpacerProps, SpaceDirection } from "./types";
import styled from "styled-components";

const StyledSpacer = styled.div<SpacerProps>`
  :before {
    content: "";
    display: block;
    padding: ${({ token, align, theme }) => makeSpace(theme.spacing[token], align)};
  }
`;

const makeSpace = (token: string, align?: SpaceDirection) => {
  switch (align) {
    case "vertical":
      return `${token} 0`;
    case "horizontal":
      return `0 0 0 ${token}`;
    default:
      return `${token} 0`;
  }
};

const Spacer = ({ token, align }: SpacerProps) => {
  return <StyledSpacer token={token} align={align} />;
};

export { Spacer, makeSpace };
