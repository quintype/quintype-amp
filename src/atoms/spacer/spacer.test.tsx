import React from "react";
import { Spacer, makeSpace } from "./spacer";
import { shallow } from "enzyme";

describe("Spacer", () => {
  it("should render", () => {
    const wrapper = shallow(<Spacer token="m" />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should generate proper margin value for vertical", () => {
    const value = makeSpace("32px", "vertical");
    expect(value).toBe("32px 0");
  });
  it("should generate proper margin value for horizontal", () => {
    const value = makeSpace("32px", "horizontal");
    expect(value).toBe("0 32px");
  });
  it("should return vertical margin when no direction is passed", () => {
    const value = makeSpace("32px");
    expect(value).toBe("32px 0");
  });
});
