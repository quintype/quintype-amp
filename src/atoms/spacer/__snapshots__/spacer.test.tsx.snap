// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer should render 1`] = `
<styled.div
  data-test-id="styled-space-vertical"
  token="m"
/>
`;
